tmt/utils.py

```diff
@@ -615,19 +733,20 @@ class FinishError(GeneralError):
 #  Utilities
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-def quote(string):
+def quote(string: str) -> str:
     """ Surround a string with double quotes """
     return f'"{string}"'
 
 
-def ascii(text):
+def ascii(text: Any) -> bytes:
     """ Transliterate special unicode characters into pure ascii """
     if not isinstance(text, str):
         text = str(text)
     return unicodedata.normalize('NFKD', text).encode('ascii', 'ignore')
 
 
-def listify(data, split=False, keys=None):
+def listify(data: Union[Tuple[Any, ...], List[Any], str, Dict[Any, Any]], split: bool = False,
```

Comment:
Wow, this looks a bit crazy but we probably can't do much better :/

Reply:
I'm afraid so - if you find the time or will, feel free to check call sites of `listify()`, but IIUIC, it gets called with dictionaries, list tuples, and strings, as documented by its code. Maybe it's called with other types, that are then returned as `[data]`, but I suppose that would be reported by mypy once we run it for other files and some code would call `listify(<int>)`, raising an error - if that's the case, we can just "give up" and turn this type to `Any` and be done with it. but since I don't know enough, I tried to be conservative here